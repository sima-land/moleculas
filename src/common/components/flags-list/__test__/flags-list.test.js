import React from 'react';
import { shallow } from 'enzyme';
import FlagsList from '../index';
import Badge from '@dev-dep/ui-nucleons/badge';

describe('<FlagsList />', () => {
  describe('without props', () => {
    const flagsList = shallow(<FlagsList />);

    it('renders correctly', () => {
      expect(flagsList).toMatchSnapshot();
    });

    it('wrapper doesn`t render', () => {
      expect(flagsList.find('.wrap')).toHaveLength(1);
      expect(flagsList.find('.item')).toHaveLength(0);
    });
  });

  describe('with props', () => {
    const flags = [{
      link: '/sale/?is_label=1',
      bgColor: '#64a0eb',
      textColor: '#FFFFFF',
      fields: [
        {
          type: 'text',
          value: '-10%',
        },
      ],
    }];
    const flagsList = shallow(<FlagsList flags={flags} className='test' />);

    it('renders correctly', () => {
      expect(flagsList).toMatchSnapshot();
    });

    it('contains flags', () => {
      const $flag = flagsList.find(Badge);
      expect($flag).toHaveLength(1);
      expect($flag.prop('fields')).toEqual([{
        type: 'text',
        value: '-10%',
      }]);
    });

    it('contains additional class', () => {
      expect(flagsList.find('.wrap').prop('className')).toEqual('wrap test');
    });
  });
});
