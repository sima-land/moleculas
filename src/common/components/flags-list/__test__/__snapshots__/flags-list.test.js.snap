// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlagsList /> with props renders correctly 1`] = `
<div
  className="wrap test"
>
  <Badge
    className="item"
    color="#64a0eb"
    fields={
      Array [
        Object {
          "type": "text",
          "value": "-10%",
        },
      ]
    }
    key="0"
    link="/sale/?is_label=1"
  />
</div>
`;

exports[`<FlagsList /> without props renders correctly 1`] = `
<div
  className="wrap"
/>
`;
